src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs

```diff
@@ -28,5 +65,22 @@ public async Task<IActionResult> Details(DetailsRequest request)
             var viewModel = await _modelMapper.Map<DetailsViewModel>(request);
             return View(viewModel);
         }
+
+
+
+        [HttpGet]
+        [Route("download", Name = RouteNames.DownloadApprentices)]
+        public async Task<IActionResult> Download(long providerId, ApprenticesFilterModel filterModel)
+        {
+            var request = new GetApprenticeshipsCsvContentRequest
+            {
+                ProviderId = providerId,
+                FilterModel = filterModel
+            };
+
+            var csvFileContent = await _modelMapper.Map<byte[]>(request);
```

Comment:
I'd be tempted to create a type to hold the byte array, so that we can keep the naming convention and also the mapper can determine things like the filename (and thus remove the CurrentDateTime dependency from the controller).

Reply:
`DownloadViewModel` created which contains the FileName, ContentType and Content